### ISSUE TITLE
Csound 6.15.0 illegal instruction in libfaustcsound.dylib on Mac OS X 10.11

### ISSUE THREAD

**Issue by anthonykozar** [csound/csound#1407]
Hello,

I installed Csound 6.15.0 on Mac OS X 10.11.6 (El Capitan) with the installer package available here on GitHub (csound-MacOS_x86_64-6.15.0.dmg) and when I run the `csound` command (regardless of options) the only output I get is

```
$ csound -h

csound command: Illegal instruction

csound command: Segmentation fault
```

There was no crash log or other information to go on.  I initially assumed that the package just wasn't compatible with 10.11 but a colleague informed me that the same version works fine for him on 10.11.  So, I reinstalled the package and tried some of the other executables.  The utilities like `atsa`, `hetro`, etc. produce a stack trace along these lines:

```
$ atsa
backtrace() returned 11 addresses
0   CsoundLib64                         0x0000000100d8dae2 signal_handler + 50
1   libfaustcsound.dylib                0x0000000103c015b6 GCC_except_table378 + 89630
2   ???                                 0x00007fff6c2e310b 0x0 + 140735008354571
3   ???                                 0x00007fff6c2e3284 0x0 + 140735008354948
4   ???                                 0x00007fff6c2df8bd 0x0 + 140735008340157
5   ???                                 0x00007fff6c2df743 0x0 + 140735008339779
6   ???                                 0x00007fff6c2df9b3 0x0 + 140735008340403
7   ???                                 0x00007fff6c2d4eb0 0x0 + 140735008296624
8   ???                                 0x00007fff6c2dc308 0x0 + 140735008326408
9   libdyld.dylib                       0x00007fff8a9e679c dlopen + 59
10  CsoundLib64                         0x0000000100f0ac42 csoundLoadExternal + 258
Csound tidy up: Illegal instruction
backtrace() returned 4 addresses
0   CsoundLib64                         0x0000000100d8dae2 signal_handler + 50
1   ???                                 0x00007fff5ee7c5b8 0x0 + 140734785635768
2   libsystem_c.dylib                   0x00007fff9659a767 exit + 55
3   CsoundLib64                         0x0000000100d8dbca signal_handler + 282
Csound tidy up: Segmentation fault
```

After removing `libfaustcsound.dylib` from the `Opcodes64` directory, `csound` and most of the utilities run without errors.  I'm guessing that the "illegal instruction" error occurs with the specific CPU in my machine.  I have a pretty old iMac (20-inch, Mid 2007).  The model details are

```
  Model Name:	iMac
  Model Identifier:	iMac7,1
  Processor Name:	Intel Core 2 Duo
  Processor Speed:	2 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache:	4 MB
  Memory:	3 GB
  Bus Speed:	800 MHz
```

My colleague's machine is a MacBook with a 2.5 GHz Intel Core i5.

I can live without the Faust opcodes but any ideas why the build appears to contain CPU instructions that are incompatible with my machine? 

Another problem is that even after removing `libfaustcsound.dylib`, the `src_conv` utility fails to run with this error:

```
dyld: Library not loaded: /Users/victor/src/csound/installer/macosx/2020-08-10-192358/installer/CsoundLib64/Package_Contents/Library/Frameworks/CsoundLib64.framework/libs/libsamplerate.0.dylib
  Referenced from: /usr/local/bin/src_conv
  Reason: image not found
Trace/BPT trap: 5
```

I assume that error will affect every user of this package.

Finally, I think I saw a report on the mailing list that the Java libraries were not included in a recent release.  I can confirm that the directory `/Library/Frameworks/CsoundLib64.framework/Versions/6.0/Resources/Java` is empty on my machine after installing 6.15.0. 

Thanks for all of the work that goes into these packages! :)
Anthony

**Comment by vlazzarini** [comment]
I think the scripts are not changing the name of src_conv dependencies. I have to check. The Java thing must be because Java failed to build, I have to check that also.

**Comment by vlazzarini** [comment]
Java was building OK, but it seems a new option was added to CMake to install it somewhere else, so the script was not picking on it. The src_conv links have been fixed now. I am uploading a new installer package.

The faustopcodes problem are probably due to LLVM and the faustlib which somehow might have some incompatibilities with your system.

**Comment by anthonykozar** [comment]
I installed the new package (without removing the old one) and everything I've tried (except the `faustopcodes` plugin) appears to work now!  `src_conv` runs just fine and appears to convert sample rates as expected.  The Java libraries are now present and blue appears to be able to load and use the Csound 6 API.

Thanks for the help and the super fast turn-around on these issues, Victor! 😃

Best regards!
Anthony
